scrapy/core/scheduler.py

```diff
@@ -6,11 +6,15 @@
 from scrapy.utils.reqser import request_to_dict, request_from_dict
 from scrapy.utils.misc import load_object
 from scrapy.utils.job import job_dir
+from queuelib import RoundRobinQueue
 
 logger = logging.getLogger(__name__)
 
+def _make_file_safe(string):
+    return "".join([c if c.isalnum() or c in {'-._'} else '_' for c in string])
```

Comment:
it seems this function is lossy - won't it make some of the requests take the same slot why the shouldn't?

Reply:
Agree.  But `c.isalnum() or c in {'-._'}` handles the vast majority of domain characters and any collisions will only degrade performance.  I added a comment.